<a href="https://colab.research.google.com/github/jeffvun/Bio-Informatics-Exercises/blob/main/Practice_III.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Practice III: Linear Regression**

### **Installing and Loading Necessary Packages**

In [1]:
install.packages("ISwR")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [2]:
library("ISwR")

### **Importing Dataset**

In [3]:
data("thuesen")
attach(thuesen)
thuesen

blood.glucose short.velocity
1  15.3          1.76          
2  10.8          1.34          
3   8.1          1.27          
4  19.5          1.47          
5   7.2          1.27          
6   5.3          1.49          
7   9.3          1.31          
8  11.1          1.09          
9   7.5          1.18          
10 12.2          1.22          
11  6.7          1.25          
12  5.2          1.19          
13 19.0          1.95          
14 15.1          1.28          
15  6.7          1.52          
16  8.6            NA          
17  4.2          1.12          
18 10.3          1.37          
19 12.5          1.19          
20 16.1          1.05          
21 13.3          1.32          
22  4.9          1.03          
23  8.8          1.12          
24  9.5          1.70

### **Creating a Linear Model**

In [4]:
lm(short.velocity ~blood.glucose)


Call:
lm(formula = short.velocity ~ blood.glucose)

Coefficients:
  (Intercept)  blood.glucose  
      1.09781        0.02196  


This gives a general linear model i.e short.velocity = 1.09781 + 0.02196 (blood.glucose)

To generate the summary statistics of the above data we use the following code

In [5]:
summary(lm(short.velocity ~blood.glucose))


Call:
lm(formula = short.velocity ~ blood.glucose)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.40141 -0.14760 -0.02202  0.03001  0.43490 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept)    1.09781    0.11748   9.345 6.26e-09 ***
blood.glucose  0.02196    0.01045   2.101   0.0479 *  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.2167 on 21 degrees of freedom
  (1 observation deleted due to missingness)
Multiple R-squared:  0.1737,	Adjusted R-squared:  0.1343 
F-statistic: 4.414 on 1 and 21 DF,  p-value: 0.0479


From the above summary statistics, we pay attention to the ones in red.
We can use the created model to generated predicted values as follows:

In [6]:
predmodel <- lm(short.velocity ~blood.glucose)
predmodel


Call:
lm(formula = short.velocity ~ blood.glucose)

Coefficients:
  (Intercept)  blood.glucose  
      1.09781        0.02196  


In [7]:
fitted(predmodel)

1        2        3        4        5        6        7        8 
1.433841 1.335010 1.275711 1.526084 1.255945 1.214216 1.302066 1.341599 
       9       10       11       12       13       14       15       17 
1.262534 1.365758 1.244964 1.212020 1.515103 1.429449 1.244964 1.190057 
      18       19       20       21       22       23       24 
1.324029 1.372346 1.451411 1.389916 1.205431 1.291085 1.306459

The output corresponds to the predicted values of short.velocity given the blood.glucose values.
We determine the range of the predicted/fitted values using the code:

In [8]:
predict(predmodel, int="c")

fit      lwr      upr     
1  1.433841 1.291371 1.576312
2  1.335010 1.240589 1.429431
3  1.275711 1.169536 1.381887
4  1.526084 1.306561 1.745607
5  1.255945 1.139367 1.372523
6  1.214216 1.069315 1.359118
7  1.302066 1.205244 1.398889
8  1.341599 1.246317 1.436881
9  1.262534 1.149694 1.375374
10 1.365758 1.263750 1.467765
11 1.244964 1.121641 1.368287
12 1.212020 1.065457 1.358583
13 1.515103 1.305352 1.724854
14 1.429449 1.290217 1.568681
15 1.244964 1.121641 1.368287
17 1.190057 1.026217 1.353898
18 1.324029 1.230050 1.418008
19 1.372346 1.267629 1.477064
20 1.451411 1.295446 1.607377
21 1.389916 1.276444 1.503389
22 1.205431 1.053805 1.357057
23 1.291085 1.191084 1.391086
24 1.306459 1.210592 1.402326